scg-system/src/main/java/io/telicent/backup/services/DatasetBackupService.java

```diff
@@ -0,0 +1,276 @@
+            response.put("reason", datasetName + " does not exist");
+            response.put("success", false);
+            return response;
+        }
+        DatasetGraph dsg = dataAccessPoint.getDataService().getDataset();
+        response.set("tdb", restoreTDB(datasetName, dsg, restorePath));
+        response.set("labels", restoreLabelStore(datasetName, dsg, restorePath));
+        return response;
+    }
+
+    ObjectNode restoreTDB(String datasetName, DatasetGraph dsg, String restorePath) {
+        ObjectNode node = MAPPER.createObjectNode();
+        String tdbRestoreFile = restorePath + "/" + datasetName + "/TDB/" + datasetName + "_backup.nq.gz";
+        node.put("restorePath", tdbRestoreFile);
+        if (!checkPathExistsAndIsFile(tdbRestoreFile)) {
+            node.put("reason", "Restore file not found: " + tdbRestoreFile);
+            node.put("success", false);
+            return node;
+        }
+        try {
+            executeRestoreTDB(dsg, tdbRestoreFile);
+            node.put("success", true);
+        } catch (Exception e) {
+            node.put("reason", e.getMessage());
+            node.put("success", false);
+        }
+        return node;
+    }
+
+    @ExcludeFromJacocoGeneratedReport
+    void executeRestoreTDB(DatasetGraph dsg, String tdbRestoreFile) throws IOException {
+        try (InputStream fis = new FileInputStream(tdbRestoreFile);
+             InputStream gis = new GZIPInputStream(fis)) {
+            Txn.executeWrite(dsg, () -> {
+                dsg.clear();
+                RDFDataMgr.read(dsg, gis, NQUADS);
+            });
+        }
+    }
+
+    ObjectNode restoreLabelStore(String datasetName, DatasetGraph dsg, String restorePath) {
+        ObjectNode node = MAPPER.createObjectNode();
+        String labelRestorePath = restorePath + "/" + datasetName + "/labels/";
+        node.put("restorePath", labelRestorePath);
+        if (dsg instanceof DatasetGraphABAC abac) {
+            LabelsStore labelsStore = abac.labelsStore();
+            if (labelsStore instanceof LabelsStoreRocksDB rocksDB) {
+                if (!checkPathExistsAndIsDir(labelRestorePath)) {
+                    node.put("reason", "Restore directory not found: " + labelRestorePath);
+                    node.put("success", false);
+                } else {
+                    try {
+                        executeRestoreLabelStore(rocksDB, labelRestorePath);
+                        node.put("success", true);
+                    } catch (RuntimeException e) {
+                        node.put("reason", e.getMessage());
+                        node.put("success", false);
+                    }
+                }
+            } else {
+                node.put("reason", "No Label Store to restore (not RocksDB)");
+                node.put("success", false);
+            }
+        } else {
+            node.put("reason", "No Label Store to restore (not ABAC)");
+            node.put("success", false);
```

Comment:
Could we argue that `success=true` in this case?

I guess not as SCG operating without ABAC is a configuration smell that we're running in insecure mode

Reply:
Yeah, I was thinking perhaps "ignore" or something else. Open to suggestions.